src/ResponseFactory.php

```diff
@@ -53,20 +48,6 @@ public function render($component, $props = [])
             }
         });
 
-        $page = [
-            'component' => $component,
-            'props' => array_merge($this->sharedProps, $props),
-            'url' => Request::getRequestUri(),
-            'version' => $this->getVersion(),
-        ];
-
-        if (Request::header('X-Inertia')) {
-            return Response::json($page, 200, [
-                'Vary' => 'Accept',
-                'X-Inertia' => 'true',
-            ]);
-        }
-
-        return View::make($this->rootView, ['page' => $page]);
+        return new Response($this->rootView, $this->sharedProps, $this->getVersion(), $component, $props);
```

Comment:
Uncle bob would say: too many args to be passed

Reply:
I changed the order of the params, putting the optional ones last. I think that's totally fine for a constructor.